app/src/main/java/com/example/riden/Adapters/RideAdapter.java

```diff
@@ -177,85 +179,82 @@ protected FilterResults performFiltering(CharSequence constraint) {
 
         @Override
         protected void publishResults(CharSequence constraint, FilterResults results) {
-            rides.clear();
-            Log.i("I", "am being claled");
-            if(!((ArrayList<Ride>) results.values).isEmpty()) {
-                rides.addAll((ArrayList<Ride>) results.values);
+            if(constraint.toString().isEmpty()) {
```

Comment:
Do you need the additional check ? Can filtering return all rides if the constraint is empty ? The code here and at the start of the else block is very similar